frontend/src/stores/knowledges/actions.ts

```diff
@@ -23,8 +24,9 @@ function removeKnowledge(knowledgeId: number) {
 }
 
 async function saveKnowledge(knowledge: Knowledge): Promise<void> {
-  await post<Knowledge[]>('/knowledges', knowledge)
-  knowledges.value.push(knowledge)
+  const response = await post<Knowledge>('/knowledges', knowledge)
+  const role = roles.value.find((role) => knowledge.roleId === role.id)
```

Comment:
You can define a method inside roles/getters, like getRoleById, where you do exactly this. I checked that you are doing this find in some places. In cases like this, you should create functions to reuse code